auctionapp/src/main/java/com/internship/auctionapp/service/impl/ItemServiceImpl.java

```diff
@@ -128,6 +142,103 @@ public List<ItemDto> getBiddedOnItemsByUser(UUID bidderId) {
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public HttpStatus addNewItem(ItemRequest itemRequest, List<MultipartFile> files, UUID id) {
+        checkItemRequestValidity(itemRequest, id);
+        Item item = mapItemRequestToItem(itemRequest, id);
+
+        List<String> imageNames = uploadPhotos(id, item.getId(), files);
+        List<Image> images = new ArrayList<>();
+        try{
+            for (String imageName : imageNames) {
+                Image image = new Image();
+                image.setUrl(imageName);
+                image.setItem(item);
+                images.add(image);
+            }
+            item.setImages(images);
+
+            updateUserInfo(itemRequest, id);
+            itemRepository.save(item);
+        }catch (BadRequestException exception){
+            throw new BadRequestException("Could not save item");
+        }
```

Comment:
- Add LOGGER info with message that item was successfully saved. That message should contain information about userId and itemId.
- Add LOGGER error with message that item couldn't be saved, before throwing an exception. Logger message should include userId and maybe item name. 

This logging could be useful when uploading and downloading images so please add logs in FileStore as well.

Reply:
There is no reason for this method to return HTTP status. Please return saved item always, or null if error occured.